### PR TITLE
added adaptive_static_chunk_size

### PR DESCRIPTION
**Pull request by kmoham6**
I added adaptive_static_chunk_size to executors.

### CODE REVIEW

File: libs/core/execution/CMakeLists.txt [hunk 1]
```diff
@@ -35,6 +35,7 @@ set(execution_headers
     hpx/execution/detail/sync_launch_policy_dispatch.hpp
     hpx/execution/execution.hpp
     hpx/execution/executor_parameters.hpp
+    hpx/execution/executors/adoptive_static_chunk_size.hpp
```
Comment (hkaiser) [inline libs/core/execution/CMakeLists.txt:38]: Could you please correct the spelling to `adaptive` (everywhere)?

File: libs/core/execution/include/hpx/execution/executors/adoptive_static_chunk_size.hpp [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+            }
+            else
+            {
+                while (chunk_size * cores * 8 < num_tasks)
+                {
+                    chunk_size *= 2;
+                }
+            }
+
+            // std::cout << "chunk size: " << chunk_size << '\n';
```
Comment (hkaiser) [inline libs/core/execution/include/hpx/execution/executors/adoptive_static_chunk_size.hpp:101]: Please remove commented code if it's not needed anymore.

File: libs/core/execution/CMakeLists.txt (hunk 1 above)
Comment (kmoham6) [reply libs/core/execution/CMakeLists.txt:38]: sure ;)

File: libs/core/execution/tests/unit/executor_parameters.cpp [hunk 3]
```diff
@@ -88,7 +88,13 @@ void test_static_chunk_size()
         parameters_test(scs);
     }
 }
-
+void test_adaptive_static_chunk_size()
```
Comment (hkaiser) [inline libs/core/execution/tests/unit/executor_parameters.cpp:91]: Is this function invoked anywhere?

Comment (kmoham6) [reply libs/core/execution/tests/unit/executor_parameters.cpp:91]: problem should be solved in the new push. Thanks!

**Review by hkaiser — APPROVED** [review]
LGTM, thanks!

**Pull request merged**